help.py

```diff
@@ -41,3 +42,276 @@ def datetime_to_epoch_millis(date: datetime) -> int:
+
+    # configure logging
+    root = logging.getLogger()
+    root.setLevel(logging.DEBUG)
+    root.handlers = list()  # remove all default handlers
+    log_format = '[%(asctime)s] [%(levelname)-8s] [%(name)-36s] [%(filename)-20s:%(lineno)-4s] %(message)s'
+
+    # create logging directory if it does not exist
+    os.makedirs(logs_dir, exist_ok=True)
+
+    # create logging file handler
+    log_file_name = datetime.utcnow().strftime('%Y%m%d%H%M%S') + f'.{edr}.log'
+    handler = logging.FileHandler(os.path.join(logs_dir, log_file_name))
+    handler.setLevel(logging.DEBUG)
+    handler.setFormatter(logging.Formatter(log_format))
+    root.addHandler(handler)
+    
+    return log
+
+def check_product_args_structure(edr: str, product_args: dict) -> dict:
+    if product_args == {}:
+        return {"result": True, "edr": edr, "required_fields": "No arguments were supplied"}
+
+    product_check = [k for k, v in product_args.items()]
+    
+    if edr == "cbc":
+        required_fields = ['device_group', 'device_policy']
+    elif edr == "cbr":
+        required_fields = ['sensor_group']
+    elif edr == "s1":
+        required_fields = ['deep_visibility']
+    else:
+        return {"result": False, "edr": edr, "required_fields": required_fields}
+
+    for field in required_fields:
+        if field not in product_check:
+            return {"result": False, "edr": edr, "required_fields": required_fields}
+
+    return {"result": True, "edr": edr, "required_fields": required_fields}
+
+
+def check_credentials_structure(edr, creds: dict) -> dict:
+    # Check credential provided for the bare minimum arguments for execution.
+    # This function checks for the presence of needed arguments; it does not check for valid input.
+    if edr in ["cbc", "cbr"]:
+        return {"result": True, "edr": edr, "required_fields": "Assuming credentials file is stored in required location"}
+    
+    cred_check = [k for k, v in creds.items() if v]
+
+    if 'creds_file' and 'profile' in cred_check:
+        return {"result": True, "edr": edr, "required_fields": ["creds_file", "profile"]}
+
+    if edr == "cbc":
+        required_fields = ['url', 'token', 'org_key']
+    elif edr == "cbr":
+        required_fields = ['url', 'token']
+    elif edr == "cortex":
+        required_fields = ['api_key', 'url', 'api_key_id', 'auth_type']
+    elif edr == "dfe":
+        if 'token' in cred_check:
+            return {"result": True, "edr": edr, "required_fields": "token"}
+        else:
+            required_fields = ['tenantId', 'appId', 'appSecret']
+    elif edr == "s1":
+        state = False
+        required_fields = ['site_ids', 'account_ids', 'account_names']
+        for i in required_fields:
+            if i in cred_check:
+                state = True
+        if state:
+            required_fields = ['url', 'token']
+            for i in required_fields:
+                if i in cred_check:
+                    state = True
```

Comment:
I'm not sure I'm following the logic here - if I provide only site_ids, it will still pass even though it also needs a URL and token?